packages/wallet/contract/src/smartWallet.js

```diff
@@ -1,90 +1,273 @@
-    myAddressNameAdmin,
-    zoe,
-    board,
+  /**
+   * Invitation makers yielded by offer results
+   *
+   * @type {MapStore<number, import('./continuing').InvitationMakers>}
+   */
+  const offerToInvitationMakers = makeScalarBigMapStore('invitation makers', {
+    durable: true,
   });
+  // #endregion
 
-  const admin = E(wallet).getAdminFacet();
-
-  /** @type {Record<string, ERef<Notifier<unknown>>>} */
-  const notifierParts = {
-    contacts: E(admin).getContactsNotifier(),
-    dapps: E(admin).getDappsNotifier(),
-    /** all current issuers */
-    issuers: E(admin).getIssuersNotifier(),
-    /** the latest offer or, in case of petname change, all offers */
-    offers: E(admin).getOffersNotifier(),
-    /** most recent payment */
-    payments: E(admin).getPaymentsNotifier(),
-    /** all current purses */
-    purses: E(admin).getPursesNotifier(),
-  };
-  const mutableState = fromEntries(keys(notifierParts).map(key => [key, []]));
-  const { subscriber, publisher } = makePublishKit();
-  publisher.publish({ ...mutableState });
-  entries(notifierParts).forEach(([key, notifier]) => {
-    void observeIteration(notifier, {
-      updateState: value =>
-        publisher.publish({
-          ...assign(mutableState, { [key]: value }),
-        }),
-    });
-  });
+  // #region other state management
 
-  const marshaller = wallet.getMarshaller();
+  /**
+   * What we've been told by the Bank
+   *
+   * @type {WeakMapStore<Brand, BrandDescriptor>} */
+  const brandDescriptors = makeScalarWeakMapStore();
+  // #endregion
 
+  // #region publishing
+  // NB: state size must not grow monotonically
   const myWalletStorageNode = E(storageNode).makeChildNode(address);
-  const storedSubscriber = makeStoredSubscriber(
-    subscriber,
-    myWalletStorageNode,
+
+  /** @type {PublishKit<PursesRecord>} */
+  const pursePublishKit = makePublishKit();
+  const pursesSubscriber = makeStoredSubscriber(
+    pursePublishKit.subscriber,
+    E(myWalletStorageNode).makeChildNode('purses'),
+    marshaller,
+  );
+
+  const publishPurses = async () => {
+    const entries = Array.from(brandPurses.entries());
+    // indexed the same as brand array to merge
+    const balances = await Promise.all(
+      entries.map(([_, purse]) => E(purse).getCurrentAmount()),
+    );
+    const record = entries.map(([brand, _], i) => {
+      const desc = brandDescriptors.get(brand);
+      return { ...desc, purseBalance: balances[i] };
+    });
+    pursePublishKit.publisher.publish(record);
+  };
+
+  /** @type {PublishKit<OffersRecord>} */
+  const offerPublishKit = makePublishKit();
+  const offersSubscriber = makeStoredSubscriber(
+    offerPublishKit.subscriber,
+    E(myWalletStorageNode).makeChildNode('offers'),
     marshaller,
   );
+  /**
+   * @param {import('./offers.js').OfferStatus} status
+   */
+  const publishOfferStatus = status => {
+    const record = { [status.id]: status };
```

Comment:
why wrap `status` in another layer?

Reply:
you mean under `id`? because the commitment of our state streams is that they're the state of "something". Pushing `status` without putting it under a key means that something can flip flop from "offer 33" to "offer 44". 

What I wrote on the `OfferRecords` type is that it's a map of all offers ever, but not stored as such. Each publication is a selection from that virtual map.

I think this is right but if you disagree let's discuss at the `OffersRecord` type.